File: Tests/SMJobKitTests/SMJClientTests.m
#import "SMJTestClient.h"
#import "SMJMissingClient.h"
#import "SMJErrorTypes.h"
@import XCTest;

@interface SMJClientTests : XCTestCase
@end

@implementation SMJClientTests

- (void) testBundledVersion
{
  XCTAssertEqualObjects([SMJTestClient bundledVersion], @"0.01", @"Missing/wrong version");
  XCTAssertEqualObjects([SMJMissingClient bundledVersion], nil, @"Missing client should return nil");
}

- (void) testInstalledVersion
{
  XCTAssertEqualObjects([SMJTestClient installedVersion], nil, @"Missing client should return nil");
}

- (void) testForProblems
{
  XCTAssertNil([SMJTestClient checkForProblems], @"TestService should not have problems");
  
  NSArray* errors = [SMJMissingClient checkForProblems];
  XCTAssertEqual(errors.count, (NSUInteger)1, @"MissingService should have one error");
  NSError* error = [errors objectAtIndex:0];
  XCTAssertEqual((SMJErrorCode)error.code, SMJErrorCodeBundleNotFound, @"Missing service should be missing");
}

@end

File: Tests/SMJobKitTests/SMJClientTests.swift
//
//  SMJClientTests.swift
//  SMJobKit
//
//  Created by Ingmar Stein on 29.03.15.
//
//

import Cocoa
import XCTest
import SMJobKit

class SMJClientTests: XCTestCase {

	func testBundledVersion() {
		XCTAssert(SMJTestClient.bundledVersion! == "0.01", "Missing/wrong version")
		XCTAssert(SMJMissingClient.bundledVersion == nil, "Missing client should return nil")
	}

	func testInstalledVersion() {
		XCTAssert(SMJTestClient.installedVersion == nil, "Missing client should return nil")
	}

	func testForProblems() {
		XCTAssertNil(SMJTestClient.checkForProblems(), "TestService should not have problems")

		let errors = SMJMissingClient.checkForProblems()
		XCTAssertNotNil(errors, "MissingService should have errors")
		XCTAssertEqual(errors!.count, 1, "MissingService should have one error")
		let error = errors!.first
		XCTAssert(SMJErrorCode(rawValue:error!.code) == .BundleNotFound, "Missing service should be missing")
	}

}
